flink-connector-aws-e2e-tests/flink-connector-aws-kinesis-streams-e2e-tests/src/test/java/org/apache/flink/connector/kinesis/sink/KinesisStreamsSinkITCase.java

```diff
@@ -0,0 +1,365 @@
+    }
+
+    @Test
+    void multipleInFlightRequestsResultsInCorrectNumberOfElementsPersisted() throws Exception {
+        new Scenario()
+                .withNumberOfElementsToSend(150)
+                .withSizeOfMessageBytes(2500)
+                .withBufferMaxTimeMS(2000)
+                .withMaxInflightReqs(10)
+                .withMaxBatchSize(20)
+                .withExpectedElements(150)
+                .runScenario();
+    }
+
+    @Test
+    void nonExistentStreamNameShouldResultInFailureInFailOnErrorIsOn() {
+        testJobFatalFailureTerminatesCorrectlyWithFailOnErrorFlagSetTo(true);
+    }
+
+    @Test
+    void nonExistentStreamNameShouldResultInFailureInFailOnErrorIsOff() {
+        testJobFatalFailureTerminatesCorrectlyWithFailOnErrorFlagSetTo(false);
+    }
+
+    private void testJobFatalFailureTerminatesCorrectlyWithFailOnErrorFlagSetTo(
+            boolean failOnError) {
+        Assertions.assertThatExceptionOfType(JobExecutionException.class)
+                .isThrownBy(
+                        () ->
+                                new Scenario()
+                                        .withSinkConnectionStreamName(
+                                                "flink-test-stream-not-exists")
+                                        .withFailOnError(failOnError)
+                                        .runScenario())
+                .havingCause()
+                .havingCause()
+                .havingCause()
+                .withMessageContaining("Stream flink-test-stream-not-exists under account");
+    }
+
+    @Test
+    void veryLargeMessagesFailGracefullyWithBrokenElementConverter() {
+        Assertions.assertThatExceptionOfType(JobExecutionException.class)
+                .isThrownBy(
+                        () ->
+                                new Scenario()
+                                        .withNumberOfElementsToSend(5)
+                                        .withSizeOfMessageBytes(2500)
+                                        .withExpectedElements(5)
+                                        .withSerializationSchema(serializationSchema)
+                                        .withPartitionKeyGenerator(longPartitionKeyGenerator)
+                                        .runScenario())
+                .havingCause()
+                .havingCause()
+                .withMessageContaining(
+                        "Encountered an exception while persisting records, not retrying due to {failOnError} being set.");
+    }
+
+    @Test
+    void badEndpointShouldResultInFailureWhenInFailOnErrorIsOn() {
+        badEndpointShouldResultInFailureWhenInFailOnErrorIs(true);
+    }
+
+    @Test
+    void badEndpointShouldResultInFailureWhenInFailOnErrorIsOff() {
+        badEndpointShouldResultInFailureWhenInFailOnErrorIs(false);
+    }
+
+    private void badEndpointShouldResultInFailureWhenInFailOnErrorIs(boolean failOnError) {
+        Properties properties = getDefaultProperties();
```

Comment:
I assume this will fail when running locally? Can we implement a switch to disable when the required keys are not present?

Reply:
Yep - modified the activation profiles to be disabled by default.